components/linking_landing_page/linking_landing_page.tsx

```diff
@@ -0,0 +1,473 @@
+import mobileImg from 'images/deep-linking/deeplinking-mobile-img.png';
+import MattermostLogoSvg from 'images/logo.svg';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import CheckboxCheckedIcon from 'components/widgets/icons/checkbox_checked_icon';
+import BrowserStore from 'stores/browser_store';
+import {LandingPreferenceTypes} from 'utils/constants';
+import * as Utils from 'utils/utils';
+
+import * as UserAgent from 'utils/user_agent';
+
+type Props = {
+    defaultTheme: any;
+    desktopAppLink?: string;
+    iosAppLink?: string;
+    androidAppLink?: string;
+    siteUrl?: string;
+    siteName?: string;
+    brandImageUrl?: string;
+    enableCustomBrand: boolean;
+}
+
+type State = {
+    protocolUnsupported: boolean;
+    browserUnsupported: boolean;
+    rememberChecked: boolean;
+    redirectPage: boolean;
+    location: string;
+    nativeLocation: string;
+    brandImageError: boolean;
+    navigating: boolean;
+}
+
+export default class LinkingLandingPage extends PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        const location = window.location.href.replace('/landing#', '');
+
+        this.state = {
+            protocolUnsupported: false,
+            browserUnsupported: false,
+            rememberChecked: false,
+            redirectPage: false,
+            location,
+            nativeLocation: location.replace(/^(https|http)/, 'mattermost'),
+            brandImageError: false,
+            navigating: false,
+        };
+    }
+
+    componentDidMount() {
+        Utils.applyTheme(this.props.defaultTheme);
+        if (this.checkLandingPreferenceApp()) {
+            this.openMattermostApp();
+        }
+
+        window.addEventListener('beforeunload', this.clearLandingPreferenceIfNotChecked);
+    }
+
+    clearLandingPreferenceIfNotChecked = () => {
+        if (!this.state.navigating) {
+            BrowserStore.clearLandingPreference(this.props.siteUrl);
+        }
+        window.removeEventListener('beforeunload', this.clearLandingPreferenceIfNotChecked);
+    }
+
+    checkLandingPreferenceBrowser = () => {
+        const landingPreference = BrowserStore.getLandingPreference(this.props.siteUrl);
+        return landingPreference && landingPreference === LandingPreferenceTypes.BROWSER;
+    }
+
+    checkLandingPreferenceApp = () => {
+        const landingPreference = BrowserStore.getLandingPreference(this.props.siteUrl);
+        return landingPreference && landingPreference === LandingPreferenceTypes.MATTERMOSTAPP;
+    }
+
+    handleChecked = () => {
+        // If it was checked, and now we're unchecking it, clear the preference
+        if (this.state.rememberChecked) {
+            BrowserStore.clearLandingPreference(this.props.siteUrl);
+        }
+        this.setState({rememberChecked: !this.state.rememberChecked});
+    }
+
+    setPreference = (pref: string, clearIfNotChecked?: boolean) => {
```

Comment:
Total nitpicking, but is there maybe some other name we could use for this instead of "preference" since that usually refers to the user preferences stored on the server?

Reply:
Sure, but I can't think of one :P